design-documents/Remove-__sleep-and-__wakeup.md

```diff
@@ -0,0 +1,35 @@
+### Remove __sleep and __wakeup methods from our code
+#### Why?
+* Serializing objects requires classes to always consider serialization process
```

Comment:
Can be fixed by writing an integration test

Reply:
No, indirect serialization cannot be covered without developers remembering to update serialization tests as well as __sleep __wakeup methods